velox/functions/lib/RegistrationHelpers.h

```diff
@@ -81,6 +81,11 @@ void registerUnaryIntegralWithTReturn(const std::vector<std::string>& aliases) {
   registerFunction<T, TReturn, int64_t>(aliases);
 }
 
+template <template <class> class T, typename TReturn>
+void registerUnaryFloatingWithTReturn(const std::vector<std::string>& aliases) {
+  registerFunction<T, TReturn, double>(aliases);
```

Comment:
Do we need to register float as well, and do we need this API if it is only used limited times?

Reply:
I am wondering whether we really need to register float since if the input is fraction, the current logic will take it as double type, it should cover float cases. Does it make sense to you?

And about this function, I suppose it's necessary as I notice it may block the velox optimization when it exists in SparkSql.

Reply:
> the current logic will take it as double type, it should cover float cases

No, I don't think so. If the input is float, I assume it would fail to find the correct signature.

Reply:
Yeah, as we have aligned to align this implementation with Spark, it will support float/integer as well in next iteration.